rfc/rfc-0009-supply-chain-switches-template-on-flag.md

```diff
@@ -1,3 +1,197 @@
+template or another. Currently, such a difference can be handled by:
+
+1. Using ytt within one template to define two different objects.
+2. Creating two templates for to handle the difference and then two supply chains
+   that are the same identical except for one step that uses template A or B.
+
+The team has expressed that they prefer having separate templates (as opposed to
+maintaining ytt templates) but is concerned about the number of supply chains that
+will be needed to pursue this approach. Their hope is to keep the number of
+supply chains small, even as many small differences between workloads could lead to
+a combinatoric explosion of paths to prod.
+
+## Explanation
+
+### Current state of Cartographer
+
+Cartographer is already capable of matching a supply chain to a workload through selectors.
+With no further effort, a platform can be created that has multiple supply chains with different
+purposes or requirements. By specifying characteristics of workloads in the workload labels,
+devs can provide information to operators necessary to choose the right supply chain for
+each workload.
+
+While this behavior is currently possible, relying only on this selection would lead to a platform
+with a large number of supply chains.
+
+Workarounds, for example templates that use ytt templating to embed conditional behavior,
+would represent a code smell that this primitive is not adequate for Cartographer users.
+
+### Condition template choice on information available from the workload
+
+When creating objects of the supply chain, Cartographer has the workload definition available.
+Allowing the supply chain to leverage the workload would add additional power to the conditional.
+
+Each step/resource in a supply chain will continue to define a templateRef, which will continue to
+specify a single template kind (e.g. ClusterImageTemplate). Rather than specifying the name of a
+particular template, authors can provide `options` which is a list of options. An option is
+a `name` and a `selector`. The only selector adopted by this RFC is `MatchFields` (though other
+selectors can be added later). A MatchFields contains:
+- `key`: a path on the workload object. The path must be prefixed by `workload`.
+- `operator`: One of `In`, `NotIn`, `Exists`, `DoesNotExist`
+- `values`: Required and allowed only when the operator is either `In` or `NotIn`. A list of json values.
+  (e.g. both strings and integers are valid values)
+
+```yaml
+apiVersion: kontinue.io/v1alpha1
+kind: ClusterSupplyChain
+metadata:
+  name: responsible-ops
+spec:
+  selector:
+    app: web
+  resources:
+    - name: source-provider
+      templateRef:
+        kind: ClusterSourceTemplate
+        options:                                  # <--- a list
+        - name: git-template
+          selector:
+            matchFields:
+              - key: workload.spec.source.git       # <--- path to field in template context
```

Comment:
This is no longer a "path to field in template context" it's a "path to field in the workload". Does that change our syntax here? I kinda like the idea of keeping the `workload` prefix just in case we do want to make other aspects of the system available later on.